biorxiv/article_distances/06_biorxiv_article_distances_cosine_abstract_tester.py

```diff
@@ -0,0 +1,423 @@
+                    .reset_index()
+                    .document.values[0]
+                ),
+                "journal": row["journal"],
+                "distance": off_article_dist[
+                    idx, remaining_journal_mapper[row["journal"]]
+                ],
+            }
+        )
+
+final_abstract_df = articles_distance_abstract_df.assign(
+    label="pre_vs_published"
+).append(pd.DataFrame.from_records(data).assign(label="pre_vs_random"))
+final_abstract_df.head()
+
+final_abstract_df = biorxiv_journal_df[["document", "preprint_doi"]].merge(
+    final_abstract_df
+)
+final_abstract_df.to_csv(
+    "output/annotated_links/article_distances_abstract_only.tsv", sep="\t", index=False
+)
+final_abstract_df.head()
+
+# # Distribution plot
+
+g = (
+    p9.ggplot(
+        final_original_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+g = (
+    p9.ggplot(
+        final_abstract_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+# # ROC-Curve to do a Global Match
+
+data_rows = []
+for df, model_label in zip(
+    [final_original_df, final_abstract_df], ["Full Text", "Abstract Only"]
+):
+    for k in tqdm.tqdm(range(1, 34503, 200)):
+        recall = recall_score(
+            df.sort_values("distance").iloc[0:k].label.tolist(),
+            ["pre_vs_published"] * k
+            if k <= df.shape[0]
+            else ["pre_vs_published"] * df.shape[0],
+            pos_label="pre_vs_published",
+        )
+
+        precision = precision_score(
+            df.sort_values("distance").iloc[0:k].label.tolist(),
+            ["pre_vs_published"] * k
+            if k <= df.shape[0]
+            else ["pre_vs_published"] * df.shape[0],
+            pos_label="pre_vs_published",
+        )
+
+        data_rows.append(
+            {"recall": recall, "precision": precision, "N": k, "model": model_label}
+        )
+
+plot_df = pd.DataFrame.from_records(data_rows)
+plot_df.head()
+
+g = (
+    p9.ggplot(plot_df, p9.aes(x="N", y="recall", color="model"))
+    + p9.geom_point()
+    + p9.labs(x="Top N predictions", y="Recall")
+)
+g.save("output/figures/abstract_vs_full_text_top_k_recall.png", dpi=600)
```

Comment:
Having a auPRC of 1 for the top 10k predictions is pretty hard to argue with (I know that's not the point of these figures, but it is striking)